api/fees/data.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math"
+	"math/big"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/cache"
+	"github.com/vechain/thor/v2/chain"
+)
+
+func newFeesData(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedSize uint32) *FeesData {
+	cacheSize := int(math.Min(float64(backtraceLimit), float64(fixedSize)))
+	maxBacktraceLimit := uint32(math.Max(float64(backtraceLimit), float64(cacheSize)))
+	return &FeesData{
+		repo:              repo,
+		cache:             cache.NewPrioCache(cacheSize),
+		bft:               bft,
+		cacheSize:         uint32(cacheSize),
+		maxBacktraceLimit: maxBacktraceLimit,
+	}
+}
```

Comment:
It seems like the cache is competing with the backtrace limit.
It's fine to have fixed cache size.

```go
func newFeesData(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedSize uint32) *FeesData {
	return &FeesData{
		repo:              repo,
		cache:             cache.NewPrioCache(int(fixedSize)),
		bft:               bft,
		maxBacktraceLimit: backtraceLimit,
	}
}
```

Reply:
Done